### PR TITLE
[docs] R docs cleanup

### PR DESCRIPTION
**Pull request by StrikerRUS**
Result: https://lightgbm.readthedocs.io/en/r_examples/R/reference/index.html

- Fixed duplicated output from examples.

Before:

image

After:

image

- Unified rendering of `lgb.Dataset` and `lgb.Booster` names.

### CODE REVIEW

**Comment by StrikerRUS** [comment]
Unfortunately, I have no R installed on my machine, so someone should update https://github.com/microsoft/LightGBM/tree/master/R-package/man folder before merging this PR.

**Comment by jameslamb** [comment]
I can regenerate the `man/` folder right now

File: R-package/R/lgb.Dataset.R [hunk 1]
```diff
@@ -646,7 +646,7 @@ Dataset <- R6::R6Class(
 #' @param colnames names of columns
 #' @param categorical_feature categorical features
 #' @param free_raw_data TRUE for need to free raw data after construct
-#' @param info a list of information of the lgb.Dataset object
+#' @param info a list of information of the \code{lgb.Dataset} object
```
Comment (jameslamb) [inline R-package/R/lgb.Dataset.R]: good change, thank you

**Comment by StrikerRUS** [comment]
@jameslamb Thank you very much!

> A few other small things in there that seem unrelated to your changes, seems they were not regenerated in past PRs.

Yeah, changes from #2176 were not regenerated.

**Pull request merged**